### ISSUE TITLE
How to protect WUD web access

### ISSUE THREAD

**Issue by ranrinc** [fmartinou/whats-up-docker#63]
Is there a possibility to implement username/password before going to WUD?

**Comment by fmartinou** [comment]
Hi @ranrinc ,

Big news: I implemented Authentication on WUD 😃 

Actually, I applied the same logic as for the other components (watchers, triggers, registries...).
I.e. it's possible to configure at the same time 0, 1 or N authentication strategies by using env vars only.

By default, without any configuration, an `Anonymous` strategy is enabled (preserving backward compatibility for existing users).

**As a start, I currently coded Basic Auth strategy.**
And I plan to gradually add new strategies (the next one should be OpenId Connect).

The documentation is not yet published but [the markdown file is here](https://github.com/fmartinou/whats-up-docker/blob/develop/docs/configuration/authentications/basic/README.md).

Basically, enabling Basic auth looks like:
```yaml
version: '3'

services:
  whatsupdocker:
    image: fmartinou/whats-up-docker
    ...
    environment:
      - WUD_AUTH_BASIC_JOHN_USER=john
      - WUD_AUTH_BASIC_JOHN_HASH=$$apr1$$8zDVtSAY$$62WBh9DspNbUKMZXYRsjS/
      - WUD_AUTH_BASIC_JANE_USER=jane
      - WUD_AUTH_BASIC_JANE_HASH=$$apr1$$5iyu65pm$$m/6I35fjUT7.1CMnS2w9d1
```

You can give it a try on version `fmartinou/whats-up-docker:develop`

Thank you for your feedback!

![image](https://user-images.githubusercontent.com/16061231/132093495-2c527380-d57c-41de-a8eb-2710a155db28.png)

**Comment by ranrinc** [comment]
> Hi @ranrinc ,
> 
> **Big news: I implemented Authentication on WUD** 😃
> 
> Actually, I applied the same logic as for the other components (watchers, triggers, registries...).
> I.e. it's possible to configure at the same time 0, 1 or N authentication strategies by using env vars only.
> 
> By default, without any configuration, an `Anonymous` strategy is enabled (preserving backward compatibility for existing users).
> 
> **As a start, I currently coded Basic Auth strategy.**
> And I plan to gradually add new strategies (the next one should be OpenId Connect).
> 
> The documentation is not yet published but [the markdown file is here](https://github.com/fmartinou/whats-up-docker/blob/develop/docs/configuration/authentications/basic/README.md).
> 
> Basically, enabling Basic auth looks like:
> 
> ```yaml
> version: '3'
> 
> services:
>   whatsupdocker:
>     image: fmartinou/whats-up-docker
>     ...
>     environment:
>       - WUD_AUTH_BASIC_JOHN_USER=john
>       - WUD_AUTH_BASIC_JOHN_HASH=$$apr1$$8zDVtSAY$$62WBh9DspNbUKMZXYRsjS/
>       - WUD_AUTH_BASIC_JANE_USER=jane
>       - WUD_AUTH_BASIC_JANE_HASH=$$apr1$$5iyu65pm$$m/6I35fjUT7.1CMnS2w9d1
> ```
> 
> You can give it a try on version `fmartinou/whats-up-docker:develop`
> 
> Thank you for your feedback!
> 
> ![image](https://user-images.githubusercontent.com/16061231/132093495-2c527380-d57c-41de-a8eb-2710a155db28.png)

WOW.... let me test it this weekend. Thanks man

**Comment by ranrinc** [comment]
It's working like a charm, hope this will become the next release. Great work bro.. and Thank you for the hard work.

**Comment by fmartinou** [comment]
Glad to know!

Yes, it will be part of the next release along with a new feature allowing to provide a link to the changelog of the container to be updated.

The release `5.4.0` will likely be available tomorrow.

**Comment by fmartinou** [comment]
`5.4.0` is out... Have fun 😃 .

**Comment by ranrinc** [comment]
yeah... again thank you soo much

**Comment by fmartinou** [comment]
FYI,
Openid Connect has been added in version `5.5.0`

**Comment by ranrinc** [comment]
> Openid

need more info.. I guess I don't know anything about opened connect.

**Comment by fmartinou** [comment]
Openid connect is a standard that allows to manage authentication on one place instead of managing it in all apps.

An openid Connect provider is needed to configure authentication once (password strategy, renewal, second factor, LDAP connection...).

There are famous Openid Connect providers  as a service  (Auth0, Okta, Google, GitHub...) and on premise as well (Keycloak, Gluu, Authelia...).

So now WUD can be integrated with any of them (any provider supporting Openid connect protocol actually).

It's a central piece for SSO (single sign on).

**Comment by ranrinc** [comment]
I guess will need your documentation on how to implement google or github to WUD hahahahaha...
